packages/fluentui/react-bindings/src/compose/compose.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import { ComposeInputComponent, ComposeInputOptions, ComposePreparedComponent, ComposePreparedOptions } from './types';
+
+export const COMPOSE_PROP = 'FLUENT_COMPOSE_CONFIG';
+
+function computeDisplayNames(InputComponent: ComposeInputComponent, options: ComposeInputOptions<any, any, any>): string[] {
+  if (options.overrideStyles) {
+    return [options.displayName || InputComponent.displayName].filter(Boolean) as string[];
+  }
+
+  // To support styles composition we need to properly pick up display names
+  const previousOptions = InputComponent[COMPOSE_PROP];
+
+  if (previousOptions) {
+    return [...previousOptions.displayNames, options.displayName].filter(Boolean) as string[];
+  }
+
+  return [InputComponent.displayName, options.displayName].filter(Boolean) as string[];
+}
+
+function compose<OverrideProps, BehaviorProps, StylesProps, ComponentProps = {}>(
+  InputComponent: ComposeInputComponent<ComponentProps>,
+  options: ComposeInputOptions<ComponentProps & OverrideProps, BehaviorProps, StylesProps> = {}
+): ComposePreparedComponent<ComponentProps & OverrideProps> {
+  const ComposedComponent: React.FunctionComponent<ComponentProps & OverrideProps> & {
+    [COMPOSE_PROP]: ComposePreparedOptions;
+  } = InputComponent.bind(null);
+
+  const { handledProps = [], mapPropsToBehavior, mapPropsToStyles } = options;
```

Comment:
if you pass in a composed component, where is the merging happening? I am not seeing it

Reply:
Few lines below: https://github.com/OfficeDev/office-ui-fabric-react/pull/12239/files#diff-ad68020a8eafa54ca1569743fb0a40b2R53